src/video_core/renderer_vulkan/vk_pipeline_cache.cpp

```diff
@@ -205,24 +205,39 @@ std::array<Shader*, Maxwell::MaxShaderProgram> VKPipelineCache::GetShaders() {
     return last_shaders = shaders;
 }
 
-VKGraphicsPipeline& VKPipelineCache::GetGraphicsPipeline(const GraphicsPipelineCacheKey& key) {
+VKGraphicsPipeline* VKPipelineCache::GetGraphicsPipeline(
+    const GraphicsPipelineCacheKey& key, VideoCommon::Shader::AsyncShaders& async_shaders) {
     MICROPROFILE_SCOPE(Vulkan_PipelineCache);
 
     if (last_graphics_pipeline && last_graphics_key == key) {
-        return *last_graphics_pipeline;
+        return last_graphics_pipeline;
     }
     last_graphics_key = key;
 
+    if (device.UseAsynchronousShaders()) {
+        std::unique_lock lock{pipeline_cache};
+        const auto [pair, is_cache_miss] = graphics_cache.try_emplace(key);
+        if (is_cache_miss) {
+            LOG_INFO(Render_Vulkan, "Compile 0x{:016X}", key.Hash());
+            const auto [program, bindings] = DecompileShaders(key.fixed_state);
+            async_shaders.QueueVulkanShader(this, bindings, program, key.renderpass_params,
+                                            key.padding, key.shaders, key.fixed_state);
+        }
+        last_graphics_pipeline = graphics_cache.at(key).get();
```

Comment:
`pair->second` will have `graphics_cache.at(key)`. There's no need to resolve the key twice.
```suggestion
        last_graphics_pipeline = pair->second;
```